native/Makefile

```diff
@@ -373,19 +383,26 @@ endif
 
 
 ifeq ($(OS),darwin)
-XCODE_VERSION=$(shell xcodebuild -version | grep Xcode | sed 's/^Xcode \([1-9]\).*/\1/g')
+XCODE_VERSION=$(shell xcodebuild -version | grep Xcode | sed 's/^Xcode \([1-9][0-9]*\).*/\1/g')
 JAVA_INCLUDES+=-I/System/Library/Frameworks/JavaVM.framework/Headers
 DEFAULT_ARCH=$(shell arch)
 ARCH=$(shell arch)
 HOST_CONFIG=--host $(ARCH)-apple-darwin
 FFI_ENV += CC="$(CC)" CFLAGS="-arch $(ARCH) $(ISYSROOT) $(COPT) $(CDEBUG)" CPPFLAGS="$(CDEFINES)" LD="$(LD) -arch $(ARCH)"
 ALT_ARCHS=
 ifneq ($(ARCH),i386)
+ifeq ($(shell echo "int main(void) { return 0; }" | $(CC) $(LOC_CC_OPTS) -arch i386 $(CFLAGS) -x c - -o /dev/null &>/dev/null; echo $$?),0)
```

Comment:
Could you explain what this should do?

Reply:
This tries to compile a tiny binary for the given architecture. If the architecture is not supported, the check will fail.

Reply:
Reverted this. Shouldn't be necessary to run compiler. The deciding factor seems to be Xcode 10.